hxckr-org/hxckr-web
# chore: Added storybook

# Add Storybook for UI Component Development

## Description
This PR introduces Storybook to our project, enabling isolated UI component development and showcasing. Storybook will allow us to develop and test our UI components independently of the backend, which is particularly useful while our backend is still in progress.

## Changes
- Added Storybook dependencies
- Created Storybook configuration files
- Added Storybook scripts to package.json
- Created a sample story for the Navbar component
- Updated next.config.mjs to support Storybook
- Updated README.md with Storybook usage instructions

## How to Test
1. Pull this branch
2. Run `npm install` to install new dependencies
3. Run `npm run storybook`
4. Open http://localhost:6006 in your browser
5. Verify that you can see the Storybook interface and the Navbar component story

## Screenshots
<img width="1425" alt="image" src="https://github.com/user-attachments/assets/b69ab1aa-3645-4e1f-bd70-458ffcaa50b9">

## Additional Notes
- Developers should now create stories for new components to facilitate isolated development and testing
- This setup will allow frontend development to progress smoothly while the backend is still being developed
